### ISSUE TITLE
Add a test / process to ensure our releases don't break Grafana / Prometheus integrations

### ISSUE THREAD

**Issue by dkatzz** [civiform/civiform#6117]
**Description**
We noticed today that the grafana dashboard was broken.

We should have a test or another process to ensure we don't break this integration with a prod release.

**Comment by melanie-exygy** [comment]
Looked into this during my on-call shift.  I recommend that we add a health check for AWS Managed Prometheus to the e2e test github action.  Originally, I considered testing the AWS Grafana dashboard, but we currently configure that dashboard manually so it would be difficult to set the dashboard up during the e2e test.  
By testing AWS Managed Prometheus, we'd be checking that it's able to pull the metrics from our metrics endpoint.  That should catch any problems where the metrics data isn't getting through, such as in the bug described above.   
The next steps could be:
1. Deploy a dev instance to test on.
2. Might need to assign an IAM role with permissions for Prometheus.
3. Use the Prometheus API to query for the metrics json and to make sure they all have values.  To handle authentication with AWS, we could try using `awscurl` https://docs.aws.amazon.com/prometheus/latest/userguide/AMP-compatible-APIs.html.
4. If that works, that we can add a task to the e2e GH action.
5. We'll probably need enable monitoring on the e2e instance through the 'civiform_server_metrics_enabled ' environment variable.

**Comment by chelseakr-exygy** [comment]
> **Correction (edited):** an earlier version of this comment claimed the Prometheus client never appends `_total` to counter names. **That was wrong**, and I've rewritten the comment below. The client *does* append it, unless the name already ends in `_total`. I got it wrong by reasoning from the dashboard and the email metrics instead of reading a real scrape — which is, awkwardly, the exact argument for this ticket. Details corrected below.

Some evidence for this while looking at the ebean metrics for an unrelated reason, including a concrete case that I think reshapes what the test needs to assert.

**The recurrence this issue exists to prevent has already happened, and it went unnoticed for about nine months.**

The actual emitted names, from a real `/metrics` scrape in a unit test:

```
# TYPE ebean_queries_total counter
# TYPE ebean_queries_total_latency_micros_total counter
# TYPE ebean_queries_mean_latency_micros_total counter     <- before the fix below
# TYPE ebean_queries_max_latency_micros_total counter      <- before the fix below
```

The rule is that the client appends `_total` to a counter unless the name already ends in `_total`. So `ebean_queries_total` keeps its name, and `ebean_queries_total_latency_micros` gains a second suffix.

[#12193](https://github.com/civiform/civiform/pull/12193) ("Fix ebean graphs that don't have data on Grafana", merged 2025-12-12) got the mean and max panels **right**. What it broke is the three total-latency panels, by dropping `total_` from the middle of the name:

| Panel query after #12193 | Actually emitted |
|---|---|
| `ebean_queries_latency_micros_total` | `ebean_queries_total_latency_micros_total` |

"Average Query Latency" and both "Total latency micros per minute" panels have therefore been empty since December. Three panels, not six — an earlier version of this comment overstated it.

**Separately, and still real:** `ebean_queries_mean_latency_micros` and `ebean_queries_max_latency_micros` are declared as `Counter`s and populated with `.inc(metric.mean())` / `.inc(metric.max())`, so each is a running sum of a non-additive quantity — neither a mean nor a max. That's the monotonically-increasing-line symptom. Fixing it means making them gauges, which also changes their emitted names (gauges get no `_total`), so those two panel queries have to change with it.

**Why the health-check approach wouldn't catch either of these.** @melanie-exygy's plan above tests that AWS Managed Prometheus can pull from the metrics endpoint. It can, and it does — the metrics flow fine. The failures are on the query side and in metric typing. A pull health check goes green while the panels stay empty.

The two assertions that would catch this class:
- every panel's `expr` references a metric name that appears in a live `/metrics` scrape, and
- metric **types** match intent.

The first is cheap and needs no AWS — scrape `/metrics` in the e2e run, parse metric names out of the dashboard JSON's `expr` fields, and diff.

**And my own mistake here is the argument for it.** I worked out the suffix rule by comparing the dashboard against `MonitoringMetricCounters` and the email metrics, concluded the client never appends `_total`, and was wrong — `email_send_total` matched only because it already ended in `_total`. What caught it was writing a test that asserted the `# TYPE` lines and running it. Reasoning about metric names without a scrape in front of you is unreliable; that's precisely what the automated check should replace.

**One thing I couldn't resolve, and it may be the real answer here.** The dashboard in the repo is the dev one, added in #8008. Per Melanie's comment the AWS dashboard is configured **manually**, so none of the above necessarily describes what AWS Grafana shows, and nothing in version control can be tested against it. Getting the AWS dashboard provisioned as code is what makes it testable, and might be worth splitting out as the prerequisite for this ticket.

Step 5 of Melanie's list looks already done, by the way: `cloud-deploy-infra/e2e-test/civiform_config_aws_oidc.sh` sets `CIVIFORM_SERVER_METRICS_ENABLED=true`.

#14019 fixes the three panel names and switches mean/max to gauges, with a test asserting the types. It doesn't address this issue — the point of this comment is that the fix is exactly the kind of change that needs the test, since the last attempt made things worse and my first attempt at *this* one did too.

---
*Investigated and drafted with Claude Code.* The metric names above are from an actual scrape (`sbt testOnly controllers.monitoring.MetricsControllerTest`), not inferred.
